### ISSUE TITLE
Optional CSS file creation creates files with two extensions

### ISSUE THREAD

**Issue by philsward** [backdrop-contrib/devel_subthemer#25]
When using the optional additional css file creation, it creates a file such as: custom.css.css

I changed line 204 from:

`$styles[$key] = $style . '.css';`

to

`$styles[$key] = $style;`

If I were to guess, line 150 sets the filename as "custom.css" in the form field. Since it's already named with a .css extension, line 204 is adding it again. I can't tell if the line above 204 is trying to do a check to see if the .css exists and if not add it? Or if it's simply just trying to see if the file exists and the path is legit... Over my head.

Maybe the better approach is to warn the user upon save if the file doesn't have a .css extension in the form field, to add one. This might be the better UX approach over assuming things (potentially incorrectly) on the backend.

At the very least, this is a bug that needs some attention.

**additional**
I checked the .info file. It correctly adds the double extension with the bug in place and does not add the double extension with the bug removed.

**Comment by docwilmot** [comment]
Fixed, thanks

**Comment by philsward** [comment]
@docwilmot this may have been closed too early.

I haven't tested yet, but theorizing about this has made me realize that the bits I recommended removing, does infact need to be there. The module needs to explicitly add .css or .js to the end of the respective files.

Why? Security.

By removing the logic that adds the CSS, (again I haven't tested) someone could create a malicious.sh file, then using another  method or exploit, write into that file. This is just a loose example.

I think what needs to happen is first a check to see if the form field has the .css extension, and if not, add the extension. If it does have the extension, ignore it.

That way if malicious.sh is entered in the form field, it would automatically correct it to malicious.sh.css which would be a benign file.

I'm still wondering if the GUI should just do the check and present an error if a .css extension isn't found at the end of the file. This still might be the better way to prevent a file from being created in the first place that is either malicious or has an unintended name. Stop bad actors and bad UX (wrong file name) before the file is even created.
